test/unit/utils.authentication.unit.ts

```diff
@@ -40,15 +76,19 @@ describe('authentication', () => {
     expect(currentUser).to.deepEqual(expectedUser);
   });
 
-  it('throws error for invalid accesstoken', async () => {
+  it('decodeAccessToken throws error for invalid accesstoken', async () => {
     const token = 'fake';
-    try {
-      await decodeAccessToken(token, SECRET);
-      expect('throws error').to.be.true();
-    } catch (err) {
-      expect(err.message).to.equal('jwt malformed');
-    }
+    const error = new JsonWebTokenError('jwt malformed');
+    return expect(decodeAccessToken(token, SECRET)).to.be.rejectedWith(error);
   });
+
+  async function createUser() {
+    user.password = await hashAsync(user.password, 10);
```

Comment:
@jannyHou 
In the test suite, please use the smallest number that's still supported by bcrypt - we use `4` in LoopBack 3.x

The higher the number is, the longer it takes to compute the hash. Slow hash computation is good in production as it makes brute-force attacks difficult, but also impractical in tests because it can significantly slow down the test suite. We have been through this in LoopBack 3.x, let's not repeat that mistake again.

Reply:
@bajtos Thanks for sharing the experience and explain the reason behind using the smallest number! 

>In the test suite, please use the smallest number that's still supported by bcrypt 

Will do it in the next PR that extracts hash functions into utils.

Reply:
@bajtos fixed in https://github.com/strongloop/loopback4-example-shopping/pull/41/files#diff-40bd51c7c6fe0f144ef2295401976579R20